### ISSUE TITLE
Client Registration mandatory?

### ISSUE THREAD

**Issue by eriklupander** [oidc-sweden/specifications#23]
**TLDR;** Should the `oidc-sweden` profile explicitly state something regarding Dynamic OPs (per Core spec. 15.2) and thus Client Registration being mandatory or optional? 

Regarding `6. Client Registration`:

The Core OpenID specification [section 15.2](https://openid.net/specs/openid-connect-core-1_0.html#DynamicMTI) specifies that *dynamic* OpenID Providers MUST support dynamic client registration.

This profile does not (yet?) specify anything regarding wether oidc-sweden compliant OPs needs to be _Dynamic_ OpenID Providers, so my interpretation of the current oidc-sweden specification is that Dynamic Client Registration is OPTIONAL for OPs implementing the oidc-sweden profile? Is that correctly understood?

While I'm aware that the profile mainly addresses aspects _diverging_ from the Core OIDC specification, I think this is a possible case where it might make sense to have some extra clarification on whether support for Dynamic Registration is `mandatory` or `optional` to support for OPs compliant with oidc-sweden, possibly put in the context of differing onboarding of RPs at various OPs.

I recall dynamic registration was briefly discussed at the last group meeting, but I can't remember whether anything explicit was said regarding dynamic registration being mandatory or not.

**Comment by martin-lindstrom** [comment]
How about this:

The profile does not mandate OpenID Provider support for Dynamic registration of Relying Parties (clients). However, an OpenID Provider compliant with this profile MUST still be able to handle the client registration parameters specified in this section and in section 2 of \[OpenID.Registration\]. This means that if an OpenID Provider registers Relying Parties by other means, it still needs to gather and maintain this client information.

**Comment by martin-lindstrom** [comment]
See https://github.com/oidc-sweden/specifications/pull/26
